crates/commitlog/src/commitlog.rs

```diff
@@ -390,10 +390,19 @@ where
         };
         trace!("commit {} n={} version={}", commit.min_tx_offset, commit.n, version);
 
+        let max_tx_offset = commit.min_tx_offset + commit.n as u64;
+        if max_tx_offset <= from {
+            continue;
+        }
+
         let records = &mut commit.records.as_slice();
         for n in 0..commit.n {
             let tx_offset = commit.min_tx_offset + n as u64;
-            de.consume_record(version, tx_offset, records)?;
+            if tx_offset < from {
+                de.consume_record(version, tx_offset, records)?;
+            } else {
+                de.decode_record(version, tx_offset, records)?;
```

Comment:
Both consume and decode drive the visitor, but decode may allocate the txdata (and then drop it immediately). I think you just want to `continue` if the offset is smaller than `from`.

Reply:
Don't you need some way to consume the bytes from the `records` stream?

Reply:
Oh, this is going to be more of a pain than I realized. Sigh.